idmanager/idmanager.go

```diff
@@ -3,51 +3,71 @@ package idmanager
 import (
 	"context"
 	"fmt"
+	"io"
 
-	"github.com/Scalingo/go-etcd-lock/lock"
+	etcdlock "github.com/Scalingo/go-etcd-lock/lock"
 	"github.com/Scalingo/sand/config"
 	"github.com/Scalingo/sand/etcd"
 	"github.com/Scalingo/sand/store"
 	"github.com/pkg/errors"
 )
 
 type Manager interface {
-	Lock(context.Context) error
+	Lock(context.Context) (Lock, error)
 	Generate(context.Context) (int, error)
-	Unlock(context.Context) error
+}
+
+type Lock interface {
```

Comment:
Name of the interface seems odd. I would call it `Unlocker`

Reply:
Changed